packages/web3-eth-abi/src/coders/base/bool.ts

```diff
@@ -0,0 +1,52 @@
+/*
+This file is part of web3.js.
+
+web3.js is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+web3.js is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with web3.js.  If not, see <http://www.gnu.org/licenses/>.
+*/
+import { AbiError, InvalidBooleanError } from 'web3-errors';
+import { AbiParameter } from 'web3-types';
+import { toBool } from 'web3-utils';
+import { DecoderResult, EncoderResult } from '../types.js';
+import { WORD_SIZE } from '../utils.js';
+import { decodeNumber, encodeNumber } from './number.js';
+
+export function encodeBoolean(param: AbiParameter, input: unknown): EncoderResult {
+	let value;
+	try {
+		value = toBool(input);
+	} catch (e) {
+		if (e instanceof InvalidBooleanError) {
+			throw new AbiError('provided input is not valid boolean value', {
+				type: param.type,
+				value: input,
+				name: param.name,
+			});
+		}
+	}
+
+	return encodeNumber({ type: 'uint8', name: '' }, Number(value));
+}
+
+export function decodeBool(_param: AbiParameter, bytes: Uint8Array): DecoderResult<boolean> {
+	const boolBytes = bytes.subarray(0, WORD_SIZE);
```

Comment:
shouldn't encoded bool be strictly 32 bytes, and if not we should throw error? instead of getting sub

Reply:
When decoding we are going abi item by item, so in this case we are just decoding bool item and returning rest of the array for other decoders

Reply:
But to your point, this subarray is unnecessary as we are not even using it. bool is basically uint8 encoded 0 or 1. Will refactor